app/services/hathi_url.rb

```diff
@@ -20,6 +21,11 @@ def build_hathi_url(key:, id:)
       json_data = Net::HTTP.get(uri)
       hathi_data = JSON.parse(json_data)
       hathi_url(hathi_data['items'].first['htid']) if hathi_data['items'].count.positive?
+    rescue JSON::ParserError => e
+      Rails.logger.warn(e.message)
+      @retry_count += 1
+      sleep 10
+      retry if @retry_count < 10
```

Comment:
Given the traffic that Hathi might be seeing, should we have a longer wait time and fewer retries? Since this is being checked in the controller, we definitely wouldn't want to wait very long, but I do wonder if 10 rapid requests would get any different response.

Reply:
@escowles with my querying for data validation I waited 30 seconds, and always got a response on the next try.  Given this was user facing I dropped it to 10 seconds, since 30 seconds is ages for a user.  I would be willing to change either number, but thought if they are really slammed then just not showing the link might be optimal, since they may also not be able to show the user the book.